src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -300,6 +300,15 @@ var ReactCompositeComponent = {
           'componentWillRecieveProps(). Did you mean componentWillReceiveProps()?',
         this.getName() || 'A component',
       );
+
+      if (isPureComponent(Component) && typeof inst.shouldComponentUpdate !== 'undefined') {
+        warning(
+          false,
+          '%s has a method called shouldComponentUpdate(). ' + 
+            'shouldComponentUpdate should not be used when extending React.PureComponent.',
```

Comment:
Maybe include the suggestion to use `React.Component` instead? Slightly more actionable.